### PR TITLE
Feature/core 491 throw error if invalid bc domain given to data contract create

### PR DESCRIPTION
**Pull request by OmairLiaquatAli**
- [CORE-491]

[CORE-491]: https://evannetwork.atlassian.net/browse/CORE-491

### CODE REVIEW

File: src/contracts/base-contract/base-contract.spec.ts [hunk 1]
```diff
@@ -95,14 +95,22 @@ describe('BaseContract', function() {
     await ipfs.stop();
   });
 
-  it('can be created', async () => {
+  it.only('can be created', async () => {
```
Comment (karladler) [inline src/contracts/base-contract/base-contract.spec.ts]: If `.only` does what I think it does, we shouldn't use it in production code, just for developing and testing a single spec. ;)

File: src/contracts/base-contract/base-contract.ts [hunk 2]
```diff
@@ -113,6 +113,18 @@ export class BaseContract extends Logger {
     let businessCenterAddress;
     if (businessCenterDomain) {
       businessCenterAddress = await this.options.nameResolver.getAddress(businessCenterDomain);
+      if (businessCenterAddress != '0x0000000000000000000000000000000000000000') {
+        const businessCenterContract = await this.options.loader.loadContract('BusinessCenterInterface', businessCenterAddress);
+        //console.log(businessCenterContract)
```
Comment (karladler) [inline src/contracts/base-contract/base-contract.ts]:
```suggestion

        //console.log(businessCenterContract)
```

File: src/contracts/base-contract/base-contract.ts [hunk 3]
```diff
@@ -113,6 +113,18 @@ export class BaseContract extends Logger {
...
+        try{
+          const returnedContract = await this.options.executor.executeContractCall(businessCenterContract,'joinSchema');
```
Comment (karladler) [inline src/contracts/base-contract/base-contract.ts]:
```suggestion
          const returnedContract = await this.options.executor.executeContractCall(businessCenterContract, 'joinSchema');
```

File: src/contracts/base-contract/base-contract.spec.ts (hunk 1 above)
Comment (OmairLiaquatAli) [reply src/contracts/base-contract/base-contract.spec.ts]: Hahaha... Agreed... we definitely should **NOT** use it in production... :)

**Pull request merged**